packages/dropdowns.next/src/elements/combobox/Option.tsx

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { LiHTMLAttributes, forwardRef, useEffect, useMemo, useRef } from 'react';
+import PropTypes from 'prop-types';
+import mergeRefs from 'react-merge-refs';
+import AddIcon from '@zendeskgarden/svg-icons/src/16/plus-stroke.svg';
+import NextIcon from '@zendeskgarden/svg-icons/src/16/chevron-right-stroke.svg';
+import PreviousIcon from '@zendeskgarden/svg-icons/src/16/chevron-left-stroke.svg';
+import SelectedIcon from '@zendeskgarden/svg-icons/src/16/check-lg-stroke.svg';
+import { IOptionProps, OPTION_TYPE, OptionType } from '../../types';
+import useComboboxContext from '../../context/useComboboxContext';
+import { OptionContext } from '../../context/useOptionContext';
+import {
+  StyledOption,
+  StyledOptionContent,
+  StyledOptionIcon,
+  StyledOptionTypeIcon
+} from '../../views';
+import { OptionMeta } from './OptionMeta';
+import { toOption, toString } from './utils';
+
+const OptionComponent = forwardRef<HTMLLIElement, IOptionProps>(
+  ({ children, icon, isDisabled, isSelected, label, type, value, ...props }, ref) => {
+    const contextValue = useMemo(() => ({ isDisabled }), [isDisabled]);
+    const { activeValue, getOptionProps, isCompact } = useComboboxContext();
+    const isActive = value === activeValue;
+    const optionRef = useRef<HTMLLIElement>(null);
+
+    useEffect(() => {
+      if (isActive) {
+        // Scroll active option into view.
+        setTimeout(() => {
+          if (optionRef.current && optionRef.current.scrollIntoView) {
+            /* istanbul ignore next */
+            optionRef.current.scrollIntoView({ block: 'nearest' });
+          }
+        });
+      }
+    }, [isActive]);
+
+    const renderActionIcon = (iconType?: OptionType) => {
+      switch (iconType) {
+        case 'add':
+          return <AddIcon />;
+
+        case 'next':
+          return <NextIcon />;
+
+        case 'previous':
+          return <PreviousIcon />;
+
+        default:
+          return <SelectedIcon />;
+      }
+    };
+    const option = toOption({ value, label, isDisabled, isSelected });
+    const optionProps = getOptionProps({
+      option,
+      ref: mergeRefs([optionRef, ref])
+    }) as LiHTMLAttributes<HTMLLIElement>;
+
+    return (
+      <OptionContext.Provider value={contextValue}>
+        <StyledOption
+          isActive={isActive}
+          isCompact={isCompact}
+          $type={type}
```

Comment:
Since we are still supporting v4 of `styled-components` (`"styled-components": "^4.2.0 || ^5.0.0"`), we should avoid transient props.

Reply:
I need to change the package dependency to `^5.0.0` – which is not a breaking change for this new package.